core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -314,7 +314,39 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   }
   bool impl_tune_tile_size() const { return m_tune_tile_size; }
 
+  tile_type tile_size_recommended() const {
+    tile_type rec_tile_sizes = {};
+
+    for (std::size_t i = 0; i < rec_tile_sizes.size(); ++i) {
+      rec_tile_sizes[i] = tile_size_recommended(i);
+    }
+    return rec_tile_sizes;
+  }
+
```

Comment:
This is not really what we use internally, right? We don't take into account that the product of the tile sizes can be larger than the maximum tile size. Shouldn't his rather do
```
   if ((inner_direction == Iterate::Right && (i < rank - 1)) ||
            (inner_direction == Iterate::Left && (i > 0))) {
          if (m_prod_tile_dims * properties.default_tile_size <
              static_cast<index_type>(properties.max_total_tile_size)) {
            m_tile[i] = properties.default_tile_size;
          } else {
            m_tile[i] = 1;
```
?

Reply:
That's a valid point. Updated the logic for the recommended tile sizes. [efce022](https://github.com/kokkos/kokkos/pull/6839/commits/efce0227db521d870cb40fab78d98e4d58ffd01a).